addons/web/static/src/model/relational_model/dynamic_list.js

```diff
@@ -284,6 +284,7 @@ export class DynamicList extends DataPoint {
             this.model.dialog.add(AlertDialog, {
                 body: _t("No valid record to save"),
                 confirm: () => this.leaveEditMode({ discard: true }),
+                dismiss: () => this.leaveEditMode({ discard: true }),
```

Comment:
As AAB suggested, it would be nice to have a test alongside the one here: https://github.com/odoo/odoo/blob/17.0/addons/web/static/tests/views/list_view_tests.js#L1637

That instead closes the modal (escape key or cross click, does not really matter) and makes sure we reverted the changes.

I know you tested it inside the documents specific use case, but it's better to have that as part of the standard framework tests.